From 67ed0175f08f82855f9a7b38f0ab0839d176a682 Mon Sep 17 00:00:00 2001
From: Shawn Lin <shawn.lin@rock-chips.com>
Date: Wed, 4 Jan 2017 15:26:25 +0800
Subject: [PATCH] block: rename the flag of emmc_disk to is_rk_disk

flag of emmc_disk was used to indicate that if the main
disk is for rockchip specific as we need to use rk specific
partition layout. But this name is vague and pointless since
the main disk could be sdmmc as we support sdmmc bootup.
Moreover, we could support more types of storage devices, for
instance, NVMe, UFS etc. Let's rename it to better reflect what
it is here.

Change-Id: Ibdeb4a930b97767d796ddfc050ee9d4d0ddeceb7
Signed-off-by: Shawn Lin <shawn.lin@rock-chips.com>
---
 block/partitions/rk.c    | 5 ++---
 drivers/mmc/card/block.c | 4 ++--
 include/linux/genhd.h    | 3 ++-
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/block/partitions/rk.c b/block/partitions/rk.c
index 76f72002134c..93d9356a1748 100755
--- a/block/partitions/rk.c
+++ b/block/partitions/rk.c
@@ -300,9 +300,8 @@ int rkpart_partition(struct parsed_partitions *state)
 	if (n < SECTOR_1G)
 		return 0;
 
-        /* ONLY be used by eMMC-disk */
-        if (1 != state->bdev->bd_disk->emmc_disk)
-                return 0;
+	if (!state->bdev->bd_disk->is_rk_disk)
+		return 0;
 
         /* Fixme: parameter should be coherence with part table */
 	cmdline = strstr(saved_command_line, "mtdparts=");
diff --git a/drivers/mmc/card/block.c b/drivers/mmc/card/block.c
index ac96333f9788..139c4e37735a 100644
--- a/drivers/mmc/card/block.c
+++ b/drivers/mmc/card/block.c
@@ -2873,9 +2873,9 @@ static int mmc_blk_probe(struct mmc_card *card)
 #if defined(CONFIG_MMC_DW_ROCKCHIP) || defined(CONFIG_MMC_SDHCI_OF_ARASAN)
 	if (card->host->restrict_caps & RESTRICT_CARD_TYPE_EMMC) {
 		this_card = card;
-		md->disk->emmc_disk = 1;
+		md->disk->is_rk_disk = true;
 	} else {
-		md->disk->emmc_disk = 0;
+		md->disk->is_rk_disk = false;
 	}
 #endif
 
diff --git a/include/linux/genhd.h b/include/linux/genhd.h
index 740bf0be0725..75cfc6b56abb 100644
--- a/include/linux/genhd.h
+++ b/include/linux/genhd.h
@@ -202,7 +202,8 @@ struct gendisk {
 	struct request_queue *queue;
 	void *private_data;
 
-	int emmc_disk;			/* Flag for emmc devive */
+	/* Flag of rockchip specific disk: eMMC/eSD, NVMe, etc. */
+	bool is_rk_disk;
 	int flags;
 	struct device *driverfs_dev;  // FIXME: remove
 	struct kobject *slave_dir;
-- 
2.35.3

